src/Modules/Admin/Commands.Admin/GetPowerBIActivityEvents.cs

```diff
@@ -0,0 +1,124 @@
+using Microsoft.PowerBI.Common.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.PowerBI.Commands.Admin
+{
+    [Cmdlet(CmdletVerb, CmdletName)]
+    [OutputType(typeof(List<object>))]
+    public class GetPowerBIActivityEvents : PowerBIClientCmdlet
+    {
+        public const string CmdletName = "PowerBIActivityEvents";
+        public const string CmdletVerb = VerbsCommon.Get;
+        private bool validationError = false;
+
+        public GetPowerBIActivityEvents() : base() { }
+
+        public GetPowerBIActivityEvents(IPowerBIClientCmdletInitFactory init) : base(init) { }
+
+        [Parameter(Mandatory = true)]
+        public string StartDateTime { get; set; }
+
+        [Parameter(Mandatory = true)]
+        public string EndDateTime { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public string ActivityType { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public int ResultType { get; set; }
+
+        protected override void BeginProcessing()
+        {
+            base.BeginProcessing();
+            try
+            {
+                DateTime.Parse(this.StartDateTime);
+            }
+            catch
+            {
+                this.validationError = true;
+                this.Logger.ThrowTerminatingError($"{nameof(this.StartDateTime)} is not a valid DateTime.");
+            }
+
+            try
+            {
+                DateTime.Parse(this.EndDateTime);
+            }
+            catch
+            {
+                this.validationError = true;
+                this.Logger.ThrowTerminatingError($"{nameof(this.EndDateTime)} is not a valid DateTime.");
+            }
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.validationError)
+            {
+                return;
+            }
+
+            string formattedStartDateTime = $"'{this.StartDateTime}'";
+            string formattedEndDateTime = $"'{this.EndDateTime}'";
+            string formattedFilter = this.ActivityType;
+            if (!string.IsNullOrEmpty(this.ActivityType))
+            {
+                formattedFilter = $"Activity eq '{this.ActivityType}'";
+            }
+
+            var finalResult = this.ExecuteCmdletHelper(formattedStartDateTime, formattedEndDateTime, formattedFilter);
+            this.LogResult(finalResult);
+        }
+
+        private List<object> ExecuteCmdletHelper(string formattedStartDateTime, string formattedEndDateTime, string formattedFilter)
+        {
+            var finalResult = new List<object>();
+            using (var client = this.CreateClient())
+            {
+                ActivityEventResponse response = client.Admin.GetPowerBIActivityEvents(formattedStartDateTime, formattedEndDateTime, null, formattedFilter);
+                while (response.ContinuationToken != null)
+                {
+                    finalResult = finalResult.Concat(response.ActivityEventEntities).ToList();
```

Comment:
Concat() has more has better performance over AddRange() but because you are calling ToList() it will actually have worse performance and extra memory allocation if you just had used AddRange().
So I suggest this is reworked to use AddRange() instead

Reply:
Removed List and made it IEnumerable, so that we can still have the Concat.